code/modules/html_interface/map/station_map.dm

```diff
@@ -80,3 +87,64 @@
 					canvas.DrawBox("#66666699", i, r)
 
 	centcommMiniMaps[filter] = canvas
+
+/proc/generateStationMinimap()
+	var/icon/canvas = icon('icons/480x480.dmi', "blank")
+
+	for(var/i = 1 to ((2 * world.view + 1)*WORLD_ICON_SIZE))
+		for(var/r = 1 to ((2 * world.view + 1)*WORLD_ICON_SIZE))
+			var/turf/tile = locate(i, r, map.zMainStation)
+			if(tile && tile.loc)
+				var/area/areaToPaint = tile.loc
+				if(areaToPaint.holomap_color)
+					canvas.DrawBox(areaToPaint.holomap_color, i, r)
+
+	extraMiniMaps |= HOLOMAP_EXTRA_STATIONMAPAREAS
+	extraMiniMaps[HOLOMAP_EXTRA_STATIONMAPAREAS] = canvas
+
+	var/icon/big_map = icon('icons/480x480.dmi', "stationmap")
+	var/icon/small_map = icon('icons/480x480.dmi', "blank")
+	var/icon/map_base = icon(holoMiniMaps[map.zMainStation])
+
+	small_map.Blend(map_base,ICON_OVERLAY)
+	small_map.Blend(canvas,ICON_OVERLAY)
+	small_map.Scale(32,32)
+
+	map_base.Blend("#79ff79",ICON_MULTIPLY)
+
+	big_map.Blend(map_base,ICON_OVERLAY)
+	big_map.Blend(canvas,ICON_OVERLAY)
+
+	//Adding Bar Marker
+	for(var/area/crew_quarters/bar/BAR in areas)
```

Comment:
>Definitely not

Why not? This is a bad way of handling a specific kind of behaviour that people will want to extend in the future.